tests/unit_tests/test_get_decorated_nodes.py

```diff
@@ -0,0 +1,81 @@
+import ast
+from typing import Union
+
+import pytest
+
+from magniv.build.build import get_decorated_nodes
+
+TEST_FILE = """from datetime import datetime
+
+import magniv
+from magniv.core import task
+from magniv import core as mc
+
+
+@magniv.core.task(schedule="@hourly", description="Hello world, printing the time", key="goodbye_world_1")
+def hello_world():
+    print(f"Hello world, the time is {datetime.now()}")
+
+@task(schedule="@daily", description="Hello world, printing a second time", key="goodbye_world_2")
+def hello_world_2():
+    print(f"Hello world, the time is {datetime.now()}")
+
+@mc.task(schedule="@daily", description="Hello world, printing a third time", key="goodbye_world_3")
+def hello_world_3():
+    print(f"Hello world, the time is {datetime.now()}")
+
+@dummydecorator
+def dummy_task():
+    print("This is a dummy task")
+    return True
+
+def dummy_task_2():
+    print("This is a dummy task")
+    return True
+
+if __name__ == "__main__":
+    hello_world()
+"""
+
+
+@pytest.fixture
+def file(tmpdir):
+    """
+    It creates a directory called tasks, creates a file called main.py in that directory, and writes
```

Comment:
nit: It feels easier to just read these three lines of code to understand what *exactly* this function does. I think it is more useful for the docstring to be highlevel what `file` is used for and when?

Reply:
very fair criticism - this actually isn't my writing, I just stole it from the one place it existed before and put it everywhere `file` was used, but per your other comments I'll condense to a single definition of `file` and per this comment will refocus the docstring used there.